packages/next/src/generators/application/files/app/page.tsx__tmpl__

```diff
@@ -0,0 +1,28 @@
+<% if (styledModule && styledModule !== 'styled-jsx') {
+  var wrapper = 'StyledPage';
+%>import styled from '<%= styledModule %>';<% } else {
+  var wrapper = 'div';
+%>
+  <%- style !== 'styled-jsx' ? `import styles from './page.module.${style}';` : '' %>
+<% }
+%>
+
+<% if (styledModule && styledModule !== 'styled-jsx') { %>
+const StyledPage = styled.div`<%- pageStyleContent %>`;
+<% }%>
+
+export function Index() {
```

Comment:
Let's make it an async function since it is a server component. Users can do any async data fetch needed within the function.